### PR TITLE
[Docs] add rpc function document

### PR DESCRIPTION
**Pull request by stalary**
## Problem Summary:

add rpc function document

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by stalary** [comment]
At present, I have only written the Chinese document, and the English document will be supplemented after review

**Comment by stalary** [comment]
TBR @yangzhg

File: docs/en/extending-doris/udf/user-defined-function-native.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-# User Define Function
+# User Define Function Native
```
Comment (jackwener) [inline docs/en/extending-doris/udf/user-defined-function-native.md:27]: UDF is a terminology, so `Native` isn't necessary

File: docs/en/extending-doris/udf/user-defined-function-native.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 {
-    "title": "User Defined Function",
+    "title": "User Defined Function Native",
```
Comment (jackwener) [inline docs/en/extending-doris/udf/user-defined-function-native.md:3]: Same problem

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+- fnCall：用于编写计算逻辑
+- checkFn：用于创建 UDF 时校验，校验函数名/参数/返回值等是否合法
+- handShake：用于接口探活
+
+## 创建 UDF
+
+目前暂不支持 UDAF 和 UDTF
+
+```
+CREATE FUNCTION
```
Comment (jackwener) [inline docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:66]: Format can be polished

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    - match_type：匹配类型
+  - PCheckFunctionResponse
+    - status：状态，0代表正常
+
+### 生成接口
+
+通过 protoc 生成代码，具体参数通过 protoc -h 查看
+
+### 实现接口
+共需要实现以下三个方法
```
Comment (jackwener) [inline docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:56]: format to be uniform. It need a new line here

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  "TYPE"="RPC"
+);
+```
+
+## 使用 UDF
+
+用户使用 UDF 必须拥有对应数据库的 `SELECT` 权限。
+
+UDF 的使用与普通的函数方式一致，唯一的区别在于，内置函数的作用域是全局的，而 UDF 的作用域是 DB内部。当链接 session 位于数据内部时，直接使用 UDF 名字会在当前DB内部查找对应的 UDF。否则用户需要显示的指定 UDF 的数据库名字，例如 `dbName`.`funcName`。
+
```
Comment (jackwener) [inline docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:93]: Redundant new line.

**Comment by jackwener** [comment]
Additional, please add the English doc in the same way.

File: docs/.vuepress/sidebar/zh-CN.js [hunk 6]
```diff
@@ -256,7 +256,8 @@ module.exports = [
         directoryPath: "udf/",
         children: [
           "contribute-udf",
-          "user-defined-function",
+          "user-defined-function-native",
```
Comment (yangzhg) [inline docs/.vuepress/sidebar/zh-CN.js:259]:
```suggestion
          "user-defined-function-cpp",
```

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+CREATE FUNCTION 
+	name ([,...])
+	[RETURNS] rettype
+	PROPERTIES (["key"="value"][,...])
+	
+```
+说明：
+
+1. PROPERTIES中`symbol`表示的是 rpc 调用传递的方法名，这个参数是必须设定的。
+2. PROPERTIES中`object_file`表示的 rpc 服务地址，目前仅支持单个地址，这个参数是必须设定的。
```
Comment (yangzhg) [inline docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:75]:
```suggestion
2. PROPERTIES中`object_file`表示的 rpc 服务地址，目前仅支持单个地址和brpc 兼容格式的集群地址，集群连接方式 参考 [格式说明](https://github.com/apache/incubator-brpc/blob/master/docs/cn/client.md#%E8%BF%9E%E6%8E%A5%E6%9C%8D%E5%8A%A1%E9%9B%86%E7%BE%A4)
```

**Comment by yangzhg** [comment]
Please add an English version

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md (hunk 3 above)
Comment (stalary) [reply docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:66]: done

File: docs/en/extending-doris/udf/user-defined-function-native.md (hunk 2 above)
Comment (stalary) [reply docs/en/extending-doris/udf/user-defined-function-native.md:3]: done

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md (hunk 5 above)
Comment (stalary) [reply docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:93]: done

File: docs/.vuepress/sidebar/en.js [hunk 8]
```diff
@@ -255,7 +255,8 @@ module.exports = [
         directoryPath: "udf/",
         children: [
           "contribute-udf",
-          "user-defined-function",
+          "user-defined-function-native",
```
[the remark this answers is not in the document]
Comment (stalary) [reply docs/.vuepress/sidebar/en.js:258]: done

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md (hunk 7 above)
Comment (stalary) [reply docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:75]: done

File: docs/.vuepress/sidebar/zh-CN.js (hunk 6 above)
Comment (stalary) [reply docs/.vuepress/sidebar/zh-CN.js:259]: done

**Comment by morningman** [comment]
I think we need a demo

**Comment by stalary** [comment]
> I think we need a demo
done

File: samples/doris-demo/udf-demo/src/main/resources/application.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+grpc:
```
Comment (morningman) [inline samples/doris-demo/udf-demo/src/main/resources/application.yml:1]: License header

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# User Defined Function Rpc
+
+可以通过 Rpc 的方式调用函数逻辑，通过 protobuf 进行数据传输，支持 Java/C++/Python/Ruby/Go/PHP/JavaScript 等多种语言
+
+## 编写 UDF 函数
```
Comment (morningman) [inline docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:31]:
```suggestion
## 编写 UDF 函数

本小节主要介绍如何开发一个 Remote RPC service。在 `samples/doris-demo/udf-demo/` 下提供了 Java 版本的示例，可供参考。
```

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# User Defined Function Rpc
```
Comment (morningman) [inline docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:27]:
```suggestion
# Remote User Defined Function Service
```

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+可以通过 Rpc 的方式调用函数逻辑，通过 protobuf 进行数据传输，支持 Java/C++/Python/Ruby/Go/PHP/JavaScript 等多种语言
```
Comment (morningman) [inline docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:29]:
```suggestion

Remote UDF Service 支持通过 RPC 的方式访问用户提供的 UDF Service，以实现用户自定义函数的执行。相比于 Native 的 UDF 实现，Remote UDF Service 有如下优势和限制：

1. 优势

    * 跨语言：可以用 Protobuf 支持的各类语言编写 UDF Service。
    * 安全：UDF 执行失败或崩溃，仅会影响 UDF Service 自身，而不会导致 Doris 进程崩溃。
    * 灵活：UDF Service 中可以调用任意其他服务或程序库类，以满足更多样的业务需求。
    
2. 使用限制

    * 性能：相比于 Native UDF，UDF Service 会带来额外的网络开销，因此性能会远低于 Native UDF。同时，UDF Service 自身的实现也会影响函数的执行效率，用户需要自行处理高并发、线程安全等问题。
    * 单行模式和批处理模式：Doris 原先的的基于行存的查询执行框架会对每一行数据执行一次 UDF RPC 调用，因此执行效率非常差，而在新的向量化执行框架下，会对每一批数据（默认2048行）执行一次 UDF RPC 调用，因此性能有明显提升。实际测试中，基于向量化和批处理方式的 Remote UDF 性能和基于行存的 Native UDF 性能相当，可供参考。
```

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md [hunk 13]
```diff
@@ -0,0 +1,98 @@
+---
+{
+    "title": "User Defined Function Rpc",
```
Comment (morningman) [inline docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:3]:
```suggestion
    "title": "Remote User Defined Function Service",
```

Also change the document file name.

File: docs/.vuepress/sidebar/en.js [hunk 14]
```diff
@@ -255,7 +255,8 @@ module.exports = [
         directoryPath: "udf/",
         children: [
           "contribute-udf",
-          "user-defined-function",
+          "user-defined-function-cpp",
```
Comment (morningman) [inline docs/.vuepress/sidebar/en.js:258]:
```suggestion
          "native-user-defined-function",
```

File: docs/.vuepress/sidebar/en.js [hunk 15]
```diff
@@ -255,7 +255,8 @@ module.exports = [
...
+          "user-defined-function-rpc",
```
Comment (morningman) [inline docs/.vuepress/sidebar/en.js:259]:
```suggestion
          "remote-user-defined-function",
```

File: samples/doris-demo/udf-demo/src/main/java/org/apache/doris/udf/FunctionGrpc.java [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * FunctionGrpc
+ *
+ * @author lirongqian
+ * @since 2022/02/08
+ */
+@GrpcService
+public class FunctionGrpc extends PFunctionServiceGrpc.PFunctionServiceImplBase {
+    
+    @Override
```
Comment (morningman) [inline samples/doris-demo/udf-demo/src/main/java/org/apache/doris/udf/FunctionGrpc.java:37]: Add comment to explain what function this is. Like the function definition, arguments type and return value type.

File: samples/doris-demo/udf-demo/src/main/java/org/apache/doris/udf/FunctionGrpc.java [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+    public void fnCall(FunctionService.PFunctionCallRequest request, StreamObserver<FunctionService.PFunctionCallResponse> responseObserver) {
+        String functionName = request.getFunctionName();
+        FunctionService.PFunctionCallResponse res;
+        if ("grpc_add".equals(functionName)) {
+            List<Types.PValues> argsList = request.getArgsList();
+            int sum = 0;
+            for (Types.PValues pValues : argsList) {
+                sum += pValues.getInt32Value(0) % 10;
```
Comment (morningman) [inline samples/doris-demo/udf-demo/src/main/java/org/apache/doris/udf/FunctionGrpc.java:45]: Why `% 10`?

Comment (stalary) [reply samples/doris-demo/udf-demo/src/main/java/org/apache/doris/udf/FunctionGrpc.java:45]: I was using it for testing, so I deleted it

File: samples/doris-demo/udf-demo/src/main/java/org/apache/doris/udf/FunctionGrpc.java (hunk 16 above)
Comment (stalary) [reply samples/doris-demo/udf-demo/src/main/java/org/apache/doris/udf/FunctionGrpc.java:37]: done

File: docs/.vuepress/sidebar/en.js (hunk 15 above)
Comment (stalary) [reply docs/.vuepress/sidebar/en.js:259]: done

File: docs/.vuepress/sidebar/en.js (hunk 14 above)
Comment (stalary) [reply docs/.vuepress/sidebar/en.js:258]: done

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md (hunk 13 above)
Comment (stalary) [reply docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:3]: done

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md (hunk 12 above)
Comment (stalary) [reply docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:29]: done

File: samples/doris-demo/udf-demo/src/main/resources/application.yml (hunk 9 above)
Comment (stalary) [reply samples/doris-demo/udf-demo/src/main/resources/application.yml:1]: done

File: docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md (hunk 11 above)
Comment (stalary) [reply docs/zh-CN/extending-doris/udf/user-defined-function-rpc.md:27]: done

**Review by hf200012 — DISMISSED** [review]
LGTM

File: docs/en/extending-doris/udf/native-user-defined-function.md [hunk 18]
```diff
@@ -1,6 +1,6 @@
 ---
 {
-    "title": "User Defined Function Cpp",
+    "title": "Native User Defined Function Service",
```
Comment (morningman) [inline docs/en/extending-doris/udf/native-user-defined-function.md:3]: This is not a "Service"

Comment (stalary) [reply docs/en/extending-doris/udf/native-user-defined-function.md:3]: done

**Review by hf200012 — APPROVED** [review]
LGTM

**Pull request merged**